managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -475,6 +476,7 @@ protected synchronized void initializeBookKeeper(final ManagedLedgerInitializeLe
         }
 
         // Calculate total entries and size
+        final List<Long> emptyLedgersToBeDeleted = new ArrayList<>();
```

Comment:
Is there a reason we are not using a thread safe list here? I think we might need one, though I have not read through all of this code.

Reply:
> Is there a reason we are not using a thread-safe list here?

It will not be read or written by multiple threads. And It has been changed to a thread-safe list to avoid future problems when others modify the code